packages/frontend/src/components/SavedDashboards/AddTilesToDashboardModal.tsx

```diff
@@ -1,118 +1,198 @@
 
     useEffect(() => {
-        if (data && mutate && isIdle) {
-            mutate({
-                tiles: [
-                    ...data.tiles,
-                    {
-                        uuid: uuid4(),
-                        type: DashboardTileTypes.SAVED_CHART,
-                        properties: {
-                            savedChartUuid: savedChart.uuid,
-                        },
-                        ...getDefaultChartTileSize(savedChart.chartConfig.type),
-                    },
-                ],
-                filters: data.filters,
-            });
+        if (dashboards && !dashboardData && !isLoading) {
+            if (dashboards.length > 0) setSelectedDashboard(dashboards[0].uuid);
         }
-    }, [data, mutate, isIdle, savedChart]);
-
-    useEffect(() => {
-        if (isSuccess) {
-            setCompletedMutation(true);
+        // If no dashboards, we always show the ""
+        if (
+            dashboards &&
+            dashboards.length == 0 &&
+            !isLoading &&
+            !showNewDashboardInput
+        ) {
+            setShowNewDashboardInput(true);
         }
-    }, [isSuccess]);
+    }, [dashboards, isLoading]);
 
     return (
-        <ActionModal
-            title="Add chart to dashboard"
-            confirmButtonLabel="Add"
-            useActionModalState={[actionState, setActionState]}
-            isDisabled={false}
-            onSubmitForm={onSubmitForm}
-            completedMutation={completedMutation}
-            ModalContent={Form}
+        <Dialog
+            isOpen={isOpen}
             onClose={onClose}
-        />
+            lazy
+            title="Add chart to dashboard"
+        >
+            <form>
+                <div className={Classes.DIALOG_BODY}>
+                    {!showNewDashboardInput && (
+                        <>
+                            <p>
+                                <b>Select a dashboard</b>
+                            </p>
+                            <HTMLSelect
+                                id="select-dashboard"
+                                fill={true}
+                                value={selectedDashboard}
+                                style={{ marginBottom: 15 }}
+                                onChange={(e) =>
+                                    setSelectedDashboard(e.currentTarget.value)
+                                }
+                                options={
+                                    dashboards
+                                        ? dashboards.map((dashboard) => ({
+                                              value: dashboard.uuid,
+                                              label: dashboard.name,
+                                          }))
+                                        : []
+                                }
+                            />
+                            <div
+                                style={{ cursor: 'pointer' }}
```

Comment:
I would prefer if we move this into an styled component, i understand why it is easier when it is just a 1 liner style, however one of the disadvantages of having inline style is that it can affect your page’s size and download time. it can also start looking messier and it is harder to maintain, as you might miss it when needing to update, and this will always override any other style.

We should only use inline styles when we don't have access to an element's stylesheet and we are trying to overwrite something.